Mathlib/Analysis/Calculus/Taylor.lean

```diff
@@ -351,3 +351,73 @@ theorem exists_taylor_mean_remainder_bound {f : ℝ → E} {a b : ℝ} {n : ℕ}
   rw [div_mul_eq_mul_div₀]
   refine taylor_mean_remainder_bound hab hf hx fun y => ?_
   exact (hf.continuousOn_iteratedDerivWithin rfl.le <| uniqueDiffOn_Icc h).norm.le_sSup_image_Icc
+
+/-! ### Extension of Taylor's Theorem to `x < x₀` -/
+
+
+theorem taylorCoeffWithin_neg {f : ℝ → ℝ} {x₀ : ℝ} (d : ℕ) :
+    taylorCoeffWithin (fun x => f (-x)) d Set.univ (-x₀)
+      = (-1)^d * taylorCoeffWithin f d Set.univ x₀ := by
+  field_simp [taylorCoeffWithin, iteratedDerivWithin_univ, iteratedDeriv_comp_neg, ←mul_assoc]
+
+theorem taylorWithinEval_neg {f : ℝ → ℝ} (x₀ x : ℝ) {n : ℕ} :
+    taylorWithinEval (fun x => f (-x)) n Set.univ (-x₀) (-x)
+      = taylorWithinEval f n Set.univ x₀ x := by
+  simp only [taylorWithinEval, taylorWithin, PolynomialModule.comp_apply,
+    PolynomialModule.map_single, PolynomialModule.eval_single, map_sum, map_neg, sub_neg_eq_add]
+  apply Finset.sum_congr rfl (fun d _ ↦ ?_)
+  simp [taylorCoeffWithin_neg d, ← mul_pow, ← mul_assoc, PolynomialModule.eval_smul,
+    Polynomial.eval_pow, Polynomial.eval_X, sub_eq_neg_add x x₀]
+
+theorem taylorCoeffWithin_eq {f : ℝ → ℝ} {x₀ : ℝ} {d : ℕ}
+    (s : Set ℝ) (hx : x₀ ∈ s) (hs : UniqueDiffOn ℝ s) (hf : ContDiff ℝ d f) :
+    taylorCoeffWithin f d s x₀ = taylorCoeffWithin f d Set.univ x₀ := by
+  simp only [taylorCoeffWithin, iteratedDerivWithin_eq_iteratedDeriv hf hs hx,
+    iteratedDerivWithin_univ]
+
+theorem taylorWithinEval_eq {f : Real → Real} {x₀ : Real} {d : Nat}
+    (s : Set Real) (hx : x₀ ∈ s) (hs : UniqueDiffOn ℝ s) (hf : ContDiff Real (d + 1) f) :
+  (taylorWithinEval f d s x₀) = (taylorWithinEval f d Set.univ x₀) := by
+  ext x; simp only [taylorWithinEval, taylorWithin]; congr 1
+  apply Finset.sum_congr rfl
+  intro k hk
+  rw [taylorCoeffWithin_eq s hx hs (ContDiff.of_le hf (by norm_cast; exact Finset.mem_range_le hk))]
+
+/-- **Taylor's theorem** with the Lagrange form of the remainder for `x < x₀`.
+
+We assume that `f` is `n+1`-times continuously differentiable on the reals.
+Then there exists an `x' ∈ Ioo x x₀` such
+that $$f(x) - (P_n f)(x₀, x) = \frac{f^{(n+1)}(x') (x - x₀)^{n+1}}{(n+1)!},$$
+where $P_n f$ denotes the Taylor polynomial of degree $n$ and $f^{(n+1)}$ is the $n+1$-th iterated
+derivative. -/
+theorem taylor_mean_remainder_lagrange' {f : ℝ → ℝ} {x x₀ : ℝ} {n : ℕ} (hx : x < x₀)
```

Comment:
I think it would be worth weakening the assumptions as little to make them match the already existing `taylor_mean_remainder_lagrange`, e.g. changing `(hf : ContDiff ℝ (n + 1) f) ` to only assume class `C^n` on the interval `[x, x₀]`.